arch/arc/core/mpu/arc_mpu.c

```diff
@@ -337,8 +601,8 @@ static void _arc_mpu_config(void)
 	u32_t r_index;
 	/*
 	 * the MPU regions are filled in the reverse order.
-	 * According to ARCv2 ISA, the mpu region with smaller
-	 * index has higher priority. The static background mpu
+	 * According to ARCv2 ISA, the MPU region with smaller
+	 * index has higher priority. The static background MPU
```

Comment:
The logic below for the v2 MPU region layout looks broken. r_index is initialized to the first default region number, but then initialized to 0 again in the for loop. And, the loop loops over all regions, instead of the configured ones only.